dashboard/app/models/contact_rollups_processed.rb

```diff
@@ -183,6 +207,49 @@ def self.extract_professional_learning_attended(contact_data)
     return uniq_courses.blank? ? {} : {professional_learning_attended: uniq_courses}
   end
 
+  def self.extract_roles(contact_data)
+    roles = []
+    # Contact is a teacher if they appears in any of the following tables
+    roles << 'Teacher' if
+      contact_data.dig('dashboard.users', 'user_id') ||
```

Comment:
do only teachers have a `user_id`?

Reply:
At this stage of the pipeline, yes. In the previous data collection step, we select only teacher accounts (only teacher accounts have emails). The teacher filer is  on line 129.
https://github.com/code-dot-org/code-dot-org/blob/7a99855caef4dcf160c178e7b00405980952af13/dashboard/app/models/contact_rollups_raw.rb#L124-L132